From cef0d7940ff741590c638ced909cb9e58b9d8bb0 Mon Sep 17 00:00:00 2001
From: Robert Marko <robimarko@gmail.com>
Date: Tue, 28 Dec 2021 20:59:18 +0100
Subject: [PATCH] mailbox: qcom-apcs-ipc: add IPQ8074 APSS clock controller
 support

IPQ8074 has the APSS clock controller utilizing the same register space as
the APCS, so provide acess to the APSS utilizing a child device like
IPQ6018 does as well, but just by utilizing the IPQ8074 specific APSS
clock driver.

Also, APCS register space in IPQ8074 is 0x6000 so max_register needs to be
updated to 0x5FFC.

Signed-off-by: Robert Marko <robimarko@gmail.com>
---
 drivers/mailbox/qcom-apcs-ipc-mailbox.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/drivers/mailbox/qcom-apcs-ipc-mailbox.c
+++ b/drivers/mailbox/qcom-apcs-ipc-mailbox.c
@@ -34,7 +34,7 @@ static const struct qcom_apcs_ipc_data i
 };
 
 static const struct qcom_apcs_ipc_data ipq8074_apcs_data = {
-	.offset = 8, .clk_name = NULL
+	.offset = 8, .clk_name = "qcom,apss-ipq8074-clk"
 };
 
 static const struct qcom_apcs_ipc_data msm8916_apcs_data = {
@@ -73,7 +73,7 @@ static const struct regmap_config apcs_r
 	.reg_bits = 32,
 	.reg_stride = 4,
 	.val_bits = 32,
-	.max_register = 0x1008,
+	.max_register = 0x5FFC,
 	.fast_io = true,
 };
 
